Tests/Realm.Tests/Database/AsyncTests.cs

```diff
@@ -155,5 +158,405 @@ static async Task<long> MeasureTiming(Func<Task> func)
                 }
             });
         }
+
+        [Test]
+        public void AsyncWrite_Action()
```

Comment:
We're super inconsistent about test case naming, but it's a good idea to at least be consistent within the same PR 😄 My suggestion would be to use the `Action_OptionalCondition_Expectation` pattern. For this test, this should roughly translate to `AsyncWrite_WithActionCallback_PersistsChanges`.